planet/src/main/java/com/google/planet/servlets/ItineraryServlet.java

```diff
@@ -54,7 +55,7 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
             List<Event> events = new Gson().fromJson(eventsJson, eventListType);
 
             ItineraryGenerator itineraryGenerator = new ItineraryGenerator();
-            List<ItineraryItem> itinerary = itineraryGenerator.generateItinerary(events);
```

Comment:
You're not returning the appropriate HTTP status codes here. Consider throwing an exception from your itineraryGenerator, catching it here, and returning the appropriate code